src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx

```diff
@@ -90,17 +89,43 @@ const RegisterForm: React.FunctionComponent<PropsType> = ({
   const onSubmit = async (data: RegisterFormData) => {
     try {
       setShowLoginModal(false);
-      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
-      if (venue.requiresEmailVerification)
-        await axios.get(codeCheckUrl(data.email));
-
       const auth = await signUp(data);
-      if (
-        auth.user &&
-        (venue.requiresTicketCode || venue.requiresEmailVerification)
-      ) {
+
+      if (venue.access?.includes(VenueAccessType.Codes)) {
```

Comment:
A good chunk of this code (if not all of it) seems to be copy/pasted or similar from the changes in `src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx`.

At the very least this should be refactored to share common elements rather than duplicating them, though it also raises the question as to why this is needed in multiple locations in the first place.